module/zfs/zap.c

```diff
@@ -296,16 +296,9 @@ zap_table_load(zap_t *zap, zap_table_phys_t *tbl, uint64_t idx, uint64_t *valp)
 	uint64_t blk = idx >> (bs-3);
 	uint64_t off = idx & ((1<<(bs-3))-1);
 
-	/*
-	 * Note: this is equivalent to dmu_buf_hold(), but we use
-	 * _dnode_enter / _by_dnode because it's faster because we don't
-	 * have to hold the dnode.
-	 */
-	dnode_t *dn = dmu_buf_dnode_enter(zap->zap_dbuf);
```

Comment:
After this change it looks like there's only a single remaining caller of `dmu_buf_dnode_enter()` in `receive_process_write_record()`.  Why don't we update it as well and remove the `dmu_buf_dnode_enter()` and `dmu_buf_dnode_exit()` functions.

Reply:
It seems to me that the remaining call in `receive_process_write_record()` is not even needed, since `dn->dn_datablkszsec << SPA_MINBLOCKSHIFT` should be equal to `dbp->db_size` we already have.  Changed it and removed the functions.